lib/kubeclient/common.rb

```diff
@@ -121,16 +123,21 @@ def discovery_needed?(method_sym)
 
     def handle_exception
       yield
-    rescue RestClient::Exception => e
+    rescue Faraday::Error => e
+      err_message = get_error_message(e)
+      response_code = e.response ? (e.response[:status] || e.response&.env&.status) : nil
+      error_klass = response_code == 404 ? ResourceNotFoundError : HttpError
+      raise error_klass.new(response_code, err_message, e.response)
+    end
+
+    def get_error_message(e)
       json_error_msg =
         begin
-          JSON.parse(e.response || '') || {}
-        rescue JSON::ParserError
+          JSON.parse(e.response[:body] || '') || {}
+        rescue StandardError
```

Comment:
I'm using `StandardError` because I also want to rescue `e.response` being `nil`.